### PR TITLE
[FlagGems Operator Development Competition] Add log10 operator

### PR DESCRIPTION
**Pull request by Re-Min**
### PR Category
Operator

### Type of Change
New Feature

### Description
This PR implements the `torch.log10` operator for FlagGems as part of the FlagGems Operator Development Competition.

**Implementation Details:**
- Added log10 operator in `src/flag_gems/ops/log10.py`
- Uses mathematical formula: `log10(x) = log(x) * (1/log(10))`
- Precomputed constant: `1/log(10) ≈ 0.43429448190325182765`
- Supports all float types (fp16, fp32, bfloat16) with automatic fp32 conversion for `tl.log()`
- Optimized with BLOCK=2048 and num_warps=8
- Registered operator in dispatch system
- Added comprehensive tests in `test_unary_pointwise_ops.py`
- Added benchmark in `benchmark/test_log10_perf.py`

**Test Results:**
- **Accuracy**: All tests pass with excellent precision
  - Known log10 values: 6/6 PASS (err=0.00e+00)
  - Random tensors: 4/4 PASS (max_err < 6e-08)
- **Performance (float32)**:
  - (4096, 4096): 0.93x speedup ✅ PASS
  - Note: Smaller shapes have lower speedup due to PyTorch's highly optimized log10 implementation

**Note**: PyTorch's log10 is extremely fast (likely using CUDA's native `log10f` instruction), making it challenging to achieve speedup on smaller tensors. Our implementation achieves competitive performance on large tensors (>= 4096x4096).

### Issue
- Associated with FlagGems Operator Development Competition

### Progress
- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
**Benchmark Results (float32):**
```
shape                     dtype  PyTorch(ms) FlagGems(ms)   Speedup   Result
(256, 256)              float32       0.0088       0.0449    0.1964     FAIL
(512, 512)              float32       0.0096       0.0450    0.2134     FAIL
(1024, 1024)            float32       0.0155       0.0470    0.3298     FAIL
(2048, 2048)            float32       0.0654       0.0809    0.8074     FAIL
(4096, 4096)            float32       0.2379       0.2563    0.9284     PASS
```

**Accuracy Test Results:**
```
Testing known log10 values:
log10(    1.00) = 0.000000 (expected 0.000000) err=0.00e+00 PASS
log10(   10.00) = 1.000000 (expected 1.000000) err=0.00e+00 PASS
log10(  100.00) = 2.000000 (expected 2.000000) err=0.00e+00 PASS
log10( 1000.00) = 3.000000 (expected 3.000000) err=0.00e+00 PASS
log10(    0.10) = -1.000000 (expected -1.000000) err=0.00e+00 PASS
log10(    0.01) = -2.000000 (expected -2.000000) err=0.00e+00 PASS

Testing random tensors:
shape=(16,)                max_err=5.96e-08 PASS
shape=(4, 4)               max_err=2.98e-08 PASS
shape=(8, 8)               max_err=5.96e-08 PASS
shape=(16, 16)             max_err=5.96e-08 PASS
```

Accuracy is perfect with negligible floating-point errors. Performance meets competition requirements for large tensors (4096x4096).

### CODE REVIEW

**Comment by douxetpur** [comment]
Hi, thank you for your participation in the FlagOS Open Computing Competition 🙏

After review, we have chosen to move forward with a different implementation for this operator. As a result, this PR will be closed.

We appreciate your time and contribution, and hope to see more of your submissions in the future.
